src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,11 +176,233 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
+                          PerformPatternIndexerLookup(syntax, receiverType, lookupResult, out indexerAccess, Compilation.GetSpecialType(SpecialType.System_Int32));
+            lookupResult.Free();
+            return success;
+        }
+
+        private bool TryFindRangeIndexerPatternForListPattern(
+            SyntaxNode syntax, TypeSymbol receiverType,
+            out MethodSymbol? sliceMethod,
+            out BoundIndexerAccess? indexerAccess)
+        {
+            sliceMethod = null;
+
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+            var lookupResult = LookupResult.GetInstance();
+            var success = PerformPatternIndexerLookup(syntax, receiverType, lookupResult, out indexerAccess, Compilation.GetWellKnownType(WellKnownType.System_Range)) ||
+                          TryFindImplicitRangeIndexerPattern(syntax, receiverOpt: null, receiverType, out sliceMethod, BindingDiagnosticBag.Discarded, lookupResult, ref useSiteInfo);
+            lookupResult.Free();
+            return success;
+        }
+
+        private bool PerformPatternIndexerLookup(SyntaxNode syntax, TypeSymbol receiverType, LookupResult lookupResult, out BoundIndexerAccess? valid, TypeSymbol? indexerArgumentType)
+        {
+            var diagnostics = BindingDiagnosticBag.Discarded;
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            if (indexerArgumentType is not null)
+            {
+                LookupMembersInType(
+                    lookupResult,
+                    receiverType,
+                    WellKnownMemberNames.Indexer,
+                    arity: 0,
+                    basesBeingResolved: null,
+                    LookupOptions.Default,
+                    originalBinder: this,
+                    diagnose: false,
+                    ref useSiteInfo);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    var indexerGroup = ArrayBuilder<PropertySymbol>.GetInstance();
+                    foreach (Symbol symbol in lookupResult.Symbols)
+                    {
+                        Debug.Assert(symbol.IsIndexer());
+                        indexerGroup.Add((PropertySymbol)symbol);
+                    }
+
+                    // PROTOTYPE(list-patterns) Use placeholder
+                    BoundExpression indexerArgumentExpr = new BoundImplicitReceiver(syntax, indexerArgumentType);
+                    var analyzedArguments = AnalyzedArguments.GetInstance(ImmutableArray.Create(indexerArgumentExpr));
+                    var result = BindIndexerOrIndexedPropertyAccess(
+                        syntax, receiverOpt: new BoundImplicitReceiver(syntax, receiverType),
+                        propertyGroup: indexerGroup, analyzedArguments, diagnostics, forListPattern: true);
+                    analyzedArguments.Free();
+                    if (result is BoundIndexerAccess { HasErrors: false } indexerAccess)
```

Comment:
This doesn't feel safe if we've discarded useSiteDiagnostics. HasErrors is a flag to help with suppressing cascaded diagnostics, not a definite measure of whether something bound successfully. If we're returning true from here, we need to include use site diagnostics.

Reply:
We are not binding the indexer. We're just trying the lookup for a couple of alternative pattern-based members, neither of which should report any errors.